src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/IActiveConfiguredValue.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+#pragma warning disable RS0030 // Do not used banned APIs (this wraps this API)
+
+namespace Microsoft.VisualStudio.ProjectSystem
+{
+    /// <summary>
+    ///     Provides an <see cref="UnconfiguredProject"/> access to an export from the active 
+    ///     <see cref="ConfiguredProject"/>. This is the singular version of <see cref="IActiveConfiguredValues{T}"/>.
+    /// </summary>
+    /// <remarks>
+    ///     <para>
+    ///         This replaces <see cref="ActiveConfiguredProject{T}"/> by returning <see langword="null"/>
+    ///         instead of throwing when the value does not exist.
+    ///     </para>
+    ///     <para>
+    ///         Consumers should specify a nullable type for <typeparamref name="T"/> if that import will be 
+    ///         satisfied by an export that will be applied to a particular capability.
+    ///     </para>
+    /// </remarks>
+    internal interface IActiveConfiguredValue<T>
```

Comment:
This is how I landed on the name; its replacing `ActiveConfiguredProject`, but I wanted:

- Something just as short
- Something that works with plural values (to handle IActiveConfiguredValues)

`Configured` == `Configuration/ConfiguredProject` in the CPS world.
`Value` is what we call the underlying value in the Export in MEF (see `ExportProvider.GetExportedValue`). I always felt that `Export` always looked weird on the import side so I avoided it. `Part` is describing the object behind the export (which can produce one or more exports), and not a correct description.

I think `IActiveConfiguredProjectValue` is probably a more precise name, but I do worry about the length; `IActiveConfiguredProjectValue<IAppDesignerFolderSpecialFileProvider`. Thoughts?

Reply:
Fair points. Maybe `IActiveConfigurationValue` as this sounds more like a value in a configuration rather than a value which is itself configured. Or do you think of the imported value as configured?

Can anything else be active? Is `IActiveValue<>` too generic?

Reply:
Nothing else can be active, but yes it is correct in describing the imported value as "configured". We use this term in lots of places to describe objects that live in the ConfiguredProject-scope such as `IPackageRestoreConfiguredInputDataSource`.